src/LAMMPS.jl

```diff
@@ -361,19 +361,64 @@ function extract_variable(lmp::LMP, name::String, group=nothing)
     end
 end
 
-function gather_atoms(lmp::LMP, name, T, count)
-    if T === Int32
-        dtype = 0
-    elseif T === Float64
-        dtype = 1
+"""
+    gather_atoms(lmp::LMP, name::String, T::Union{Type{Int32}, Type{Float64}}, count::Integer, ids::Union{Nothing, Array{Int32}}=nothing)
+"""
+function gather_atoms(lmp::LMP, name::String, T::Union{Type{Int32}, Type{Float64}}, count::Integer, ids::Union{Nothing, Array{Int32}}=nothing)
+    dtype = (T === Float64)
+
+    if ids === nothing
+        natoms = get_natoms(lmp)
+        data = Array{T, 2}(undef, (count, natoms))
+        API.lammps_gather_atoms(lmp, name, dtype, count, data)
+    else
+        ndata = length(ids)
+        data = Array{T, 2}(undef, (count, ndata))
+        API.lammps_gather_atoms_subset(lmp, name, dtype, count, ndata, ids, data)
+    end
+    
+    check(lmp)
+    return data
+end
+
+"""
+    gather(lmp::LMP, name::String, T::Union{Type{Int32}, Type{Float64}}, count::Integer, ids::Union{Nothing, Array{Int32}}=nothing)
+"""
+function gather(lmp::LMP, name::String, T::Union{Type{Int32}, Type{Float64}}, count::Integer, ids::Union{Nothing, Array{Int32}}=nothing)
+    dtype = (T === Float64)
+
+    if ids === nothing
+        natoms = get_natoms(lmp)
+        data = Array{T, 2}(undef, (count, natoms))
+        API.lammps_gather(lmp, name, dtype, count, data)
```

Comment:
What's the difference between `gather` and `gather_atomis`?

Reply:
It seems like `gather_atoms` is basically just a more restrictive version of `gather`. If I've understood it correctlly, `gather_atoms` only works on normal atom data, so the data you could get by calling `extract_atoms` while `gather`works on any per-atom property such as computes and fixes.

I've left `gather_atoms` in because I didn't want to remove any already existing methods

Reply:
we could use the same logic from `extract_atom` in `gather_atom` to determine the datatype and count of the property, wich would make it a more convenient version of `gather` where the datatype and count are determined through the C-API

Reply:
Yeah I think I would be okay with `@deprecate gather_atoms gather`

> where the datatype and count are determined through the C-API

One downside of this is that the resulting Julia code is type-unstable. 
So having the user provide the eltype of the output is probably a better API design,
but we should safety check that the LAMMPS eltype corresponds.